portal-impl/src/com/liferay/portlet/documentlibrary/service/impl/DLFileEntryLocalServiceImpl.java

```diff
@@ -239,8 +241,20 @@ public DLFileEntry addFileEntry(
 			PortalUtil.getCurrentAndAncestorSiteGroupIds(groupId), folderId,
 			fileEntryTypeId);
 
+		String actualExtension = null;
+
+		if (inputStream instanceof ByteArrayFileInputStream) {
+			File uploadedFile =
+				((ByteArrayFileInputStream)inputStream).getFile();
```

Comment:
The cast counts when avoiding chaining. Create a variable of type ByteArrayFileInputStream.

Reply:
Changed.